### PR TITLE
 Create the disk initialization and disk selection modules 

### PR DESCRIPTION
**Pull request by poncovka**
The disk initialization module handles kickstart commands zerombr
and clearpart. It provides API for formatting and initializing disks.

The disk selection module handler the kickstart command ignoredisk.
It provides API for selecting disks for the installation or ignoring
them by the installer.

### CODE REVIEW

File: pyanaconda/modules/common/base/base.py [hunk 1]
```diff
@@ -51,48 +85,38 @@ def run(self):
...
 
 
-class KickstartModule(BaseModule):
-    """Base implementation of a kickstart module.
-
-    Instances of this class can be published with a DBus interface
-    defined by KickstartModuleInterface.
-    """
+class KickstartBaseModule(BaseModule):
+    """Implementation of a basic kickstart module."""
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/modules/common/base/base.py]: Fixed.

File: pyanaconda/modules/storage/disk_initialization/initialization.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        self._drives_to_clear = []
+
+    def publish(self):
+        """Publish the module."""
+        DBus.publish_object(STORAGE_DISK_INITIALIZATION.object_path,
+                            DiskInitializationInterface(self))
+
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        self.set_can_format_unrecognized(data.zerombr.zerombr)
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/modules/storage/disk_initialization/initialization.py]: It is not ideal, but it is difficult to find a property name that doesn't look like a name of a method. For example, the name `check_something` suggests that it is a method that checks something.

File: pyanaconda/modules/common/base/base.py [hunk 3]
```diff
@@ -51,48 +85,38 @@ def run(self):
...
-    Instances of this class can be published with a DBus interface
-    defined by KickstartModuleInterface.
-    """
+class KickstartBaseModule(BaseModule):
+    """Implementation of a base kickstart module."""
 
     def __init__(self):
         super().__init__()
-        self._published_tasks = {}
         self._module_properties_changed = Signal()
```
Comment (rvykydal) [inline pyanaconda/modules/common/base/base.py]: Isn't this already done in BaseModule?

File: pyanaconda/modules/common/base/base.py [hunk 4]
```diff
@@ -104,6 +128,21 @@ def publish_task(self, namespace, task, message_bus=DBus):
         object_path = publish_task(message_bus, namespace, task)
         self.published_tasks[task] = object_path
 
+
+class KickstartModule(MainModule, KickstartBaseModule):
+    """Implementation of a main kickstart module.
+
+    The main kickstart module is able to porse and generate the given
```
Comment (rvykydal) [inline pyanaconda/modules/common/base/base.py]: porse -> parse

File: pyanaconda/modules/storage/disk_initialization/initialization.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+        return self._initialization_mode
+
+    def set_initialization_mode(self, mode):
+        """Set the initialization mode.
+
+        :param mode: an instance of InitializationMode
+        """
+        self._initialization_mode = mode
+        self.initialization_mode_changed.emit()
+        log.debug("The initialization mode set to '%s'.", mode)
```
Comment (rvykydal) [inline pyanaconda/modules/storage/disk_initialization/initialization.py]: set to -> is set to

File: pyanaconda/modules/storage/disk_initialization/initialization.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        self.default_disk_label_changed.emit()
+        log.debug("Default disk label is set to '%s'.", label)
+
+    @property
+    def can_format_unrecognized(self):
+        """Can be disks whose formatting is unrecognized are initialized?"""
+        return self._can_format_unrecognized
+
+    def set_can_format_unrecognized(self, value):
+        """Can be disks whose formatting is unrecognized are initialized?
```
Comment (rvykydal) [inline pyanaconda/modules/storage/disk_initialization/initialization.py]: remove "are"

File: pyanaconda/modules/storage/disk_initialization/initialization_interface.py [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    def SetDefaultDiskLabel(self, label: Str):
+        """Set the default disk label to use.
+
+        :param label: a disk label
+        """
+        self.implementation.set_default_disk_label(label)
+
+    @property
+    def CanFormatUnrecognized(self) -> Bool:
+        """Can be disks whose formatting is unrecognized are initialized?"""
```
Comment (rvykydal) [inline pyanaconda/modules/storage/disk_initialization/initialization_interface.py]: remove "are"

File: pyanaconda/modules/storage/disk_initialization/initialization_interface.py [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        return self.implementation.can_format_unrecognized
+
+    @emits_properties_changed
+    def SetCanFormatUnrecognized(self, value: Bool):
+        """Can be disks whose formatting is unrecognized are initialized?
```
Comment (rvykydal) [inline pyanaconda/modules/storage/disk_initialization/initialization_interface.py]: remove "are"

File: pyanaconda/modules/common/base/base.py [hunk 9]
```diff
@@ -104,6 +128,21 @@ def publish_task(self, namespace, task, message_bus=DBus):
...
+    kickstart string based on its kickstart specification.
+    """
+
+    def __init__(self):
+        MainModule.__init__(self)
+        KickstartBaseModule.__init__(self)
```
Comment (rvykydal) [inline pyanaconda/modules/common/base/base.py]: I'm just wondering why not to use super here as well but guess you have a good reason.

File: pyanaconda/modules/storage/disk_initialization/initialization.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkonecny12) [reply pyanaconda/modules/storage/disk_initialization/initialization.py]: I don't know but `set_can_format_unrecognized` sounds to me like a setter for a method too. There is a `can` word which means that it is doing something -> method.

File: pyanaconda/modules/storage/disk_initialization/initialization_interface.py (hunk 8 above)
Comment (poncovka) [reply pyanaconda/modules/storage/disk_initialization/initialization_interface.py]: Fixed.

File: pyanaconda/modules/storage/disk_initialization/initialization_interface.py (hunk 7 above)
Comment (poncovka) [reply pyanaconda/modules/storage/disk_initialization/initialization_interface.py]: Fixed.

File: pyanaconda/modules/storage/disk_initialization/initialization.py (hunk 6 above)
Comment (poncovka) [reply pyanaconda/modules/storage/disk_initialization/initialization.py]: Fixed.

File: pyanaconda/modules/storage/disk_initialization/initialization.py (hunk 5 above)
Comment (poncovka) [reply pyanaconda/modules/storage/disk_initialization/initialization.py]: Fixed.

File: pyanaconda/modules/common/base/base.py (hunk 4 above)
Comment (poncovka) [reply pyanaconda/modules/common/base/base.py]: Fixed.

File: pyanaconda/modules/common/base/base.py (hunk 3 above)
Comment (poncovka) [reply pyanaconda/modules/common/base/base.py]: Removed.

File: pyanaconda/modules/common/base/base.py (hunk 9 above)
Comment (poncovka) [reply pyanaconda/modules/common/base/base.py]: `super().__init__()` will call only `MainModule.__init__(self)`

Comment (poncovka) [reply pyanaconda/modules/common/base/base.py]: Fixed.

**Comment by poncovka** [comment]
Updated.

**Pull request merged**